### ISSUE TITLE
natural_earth - ocean_basins_50: mask not wrapping when lon = 0

### ISSUE THREAD

**Issue by pochedls** [regionmask/regionmask#410]
## Issue 
I think I have an edge case in which case mask data is missing at the dateline when longitude = 0 (in this case the longitude values span 0 : 360). I tried `wrap_lon` = `True`, `False`, `180`, and `360`. Is there a workaround for this? Or do you think there is a possible straightforward fix? 

Aside: Regionmask has been working great – thank you for advancing this library. 

## minimal example
```
import xarray as xr
import regionmask

# open file
fn = "http://aims3.llnl.gov/thredds/dodsC/css03_data/CMIP6/CMIP/CNRM-CERFACS/CNRM-CM6-1/historical/r1i1p1f2/Amon/ts/gr/v20180917/ts_Amon_CNRM-CM6-1_historical_r1i1p1f2_gr_185001-201412.nc" 
ds = xr.open_dataset(fn)

# create mask
ocean = regionmask.defined_regions.natural_earth_v5_0_0.ocean_basins_50
mask = ocean.mask(ds.lon, ds.lat)
mask.plot()
```
![Screen Shot 2023-04-27 at 1 11 45 PM](https://user-images.githubusercontent.com/3698109/234980834-30274f20-e7e8-4507-98e2-e38578ccd9d6.png)

**Comment by mathause** [comment]
Thanks for using regionmask! I think that is an issue of the ocean region:

```python
ocean.bounds_global
```
```
[-180.0, -85.19206064243458, 179.99989708894586, 89.99999999999996]
```

I only see a dirty fix, by slightly moving lon at 180°:

```python
import numpy as np
import regionmask
ocean = regionmask.defined_regions.natural_earth_v5_0_0.ocean_basins_50


lat = np.arange(90., -90, -1)
lon = np.arange(0., 360)

lon_orig = lon.copy()

# offset lon==180 a tiny bit to ensure all gridpoins are masked
lon[lon == 180] = 180 - 0.0002

mask = ocean.mask(lon, lat)
mask = mask.assign_coords(lon=lon_orig)

mask.plot()
```

(NOTE: this might need some changes when working with the xarray coords instead of numpy arrays)

**Comment by pochedls** [comment]
Thank you for the fast feedback – I can confirm that this fixes the issue (I realize now that the issue was with the 180<sup>o</sup> coordinate). 

I'm curious if you think a fix for this *could* be incorporated into regionmask – either by a) detecting this unique situation and handling it (with a warning) or b) getting feedback from [naturalearthdata](https://www.naturalearthdata.com/) on how to extend the bound to the 180<sup>o</sup> edge.

**Comment by mathause** [comment]
I am not against it, but I don't see how to do it reliably...

**Comment by pochedls** [comment]
Thanks for the workaround and feedback. I think I will revisit this soon, but I'd be okay with you closing this thread (given the workaround you provided). If you do think of a reliable way to deal with this, please ping me on this thread (I could attempt a PR).

**Comment by mathause** [comment]
This should not have been closed in #479. Reopening.
